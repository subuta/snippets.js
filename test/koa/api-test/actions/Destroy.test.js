/* global expect, describe, it */

import { build, format, snippets as s } from 'bld.js'

import DestroyTest from 'lib/koa/api-test/actions/Destroy'

describe('koa/api-test/actions/Destroy', () => {
  it('should Destroy koa Destroy action test', () => {
    expect(format(DestroyTest({
      model: 'user',
      config: {}
    }))).toMatchSnapshot()
  })
})
