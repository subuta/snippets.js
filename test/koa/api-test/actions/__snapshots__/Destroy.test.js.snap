// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`koa/api-test/actions/Destroy should Destroy koa Destroy action test 1`] = `
"test('delete should delete user', async (t) => {
  const {request, Article} = t.context

  let articles = await Article.query()
  t.deepEqual(articles.length, 3)

  // mock jwks
  const token = createToken(privateKey, '123', currentUser)
  jwksEndpoint('http://localhost', [{pub: publicKey, kid: '123'}])

  const response = await request
    .delete('/api/users/2054')
    .set('Authorization', \`Bearer \${token}\`)

  articles = await Article.query()
  t.deepEqual(articles.length, 2)

  t.is(response.status, 204)
  t.deepEqual(response.body, {})
})
"
`;
